discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -136,13 +137,19 @@ def refresh_user_balances(session: Session, redis: Redis, accts=List[str]):
 def process_transfer_instruction(
     session: Session,
     redis: Redis,
-    sender_account: str,
-    receiver_account: str,
+    instruction: TransactionMessageInstruction,
+    account_keys: List[str],
     meta: ResultMeta,
     tx_sig: str,
     slot: int,
     challenge_event_bus: ChallengeEventBus,
 ):
+    # The transaction might list sender/receiver in a different order in the pubKeys.
+    # The "accounts" field of the instruction has the mapping of accounts to pubKey index
```

Comment:
I think sketching out a really simplified version of this structure in a comment would be super duper helpful for future maintainers

Reply:
lets get this up and tested on stage first and maybe I can follow up with a nice explainer once we're confident in the release?